### ISSUE TITLE
Gampad on iOS exported App not recognized

### ISSUE THREAD

**Issue by bubbblebub** [Scirra/Construct-3-bugs#3557]
As Gamepads are supported now on iOS 13 I connected a XBox Controller to my Construct 3 Project. It works fine in the Browser, but not as an App, exported with Xcode

## Attach a .c3p
[iOS_Gamepad_report.zip](https://github.com/Scirra/Construct-3-bugs/files/3982461/iOS_Gamepad_report.zip)

## Steps to reproduce

1. export Project to Xcode
2. run it on attached device with iOS 13
3. make sure you have a Gamepad connected to your device. Press Button A

## Observed result

Nothing happens, Gamepad not recognized
## Expected result

Text should update to "Button A is pressed"
## More details

-I am using an I Pad, iOS 13.3
-"Gamepads are supported" does trigger
-"has gamepads" does not trigger
-I am using Xcode Version 11.3 (11C29)

**Affected browsers/platforms:** 

**First affected release:** 

## System details

<details><summary>View details</summary>

Platform information
Browser: Chrome
Browser version: 79.0.3945.88
Browser engine: Chromium
Browser architecture: (unknown)
Context: browser
Operating system: Mac OS X
Operating system version: 10.14_6
Operating system architecture: (unknown)
Device type: desktop
Device pixel ratio: 1
Logical CPU cores: 8
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.88 Safari/537.36
C3 release: r180 (beta)
Language setting: en-US

Local storage
Storage quota (approx): 59 gb
Storage usage (approx): 837 mb (1.4%)
Persistant storage: No

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: yes
Supports highp precision: yes
Vendor: ATI Technologies Inc.
Renderer: AMD Radeon Pro 560 OpenGL Engine
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 2048
Extensions:

EXT_color_buffer_float
EXT_disjoint_timer_query_webgl2
EXT_float_blend
EXT_texture_filter_anisotropic
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
Audio information
System sample rate: 44100 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

Ogg Opus (audio/ogg; codecs=opus)
WebM Opus (audio/webm; codecs=opus)
Ogg Vorbis (audio/ogg; codecs=vorbis)
WebM Vorbis (audio/webm; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by AshleyScirra** [comment]
@shortercode could you give this a spin on a test device and verify? Our gamepad code is identical between the browser and WKWebView - I suspect this issue really is an iOS bug along the lines of "gamepad API doesn't work in WKWebView", so the best thing to do is just verify it then file it with Apple.

**Comment by shortercode** [comment]
I can confirm that it doesn't appear to work. I wasn't able to find an Xbox controller to test with but a Dualshock 4 gave me the same results as OP. It's not immediately clear if this is intended, or was just overlooked. I have filed a bug on the webkit tracker which is [here](https://bugs.webkit.org/show_bug.cgi?id=205448).

A quick search shows 2 cordova plugins to add support for the gamepad API. However, they both appear to be several years old, unmaintained and relatively unused. I would be surprised if either of them worked... It's best to wait for a response from the webkit team.

Closing as external, can reopen if webkit don't show any intention of enabling the API in webviews.
